tests/imap/data.py

```diff
@@ -102,6 +103,23 @@ def __init__(self):
         self._data = {}
         self.selected_folder = None
         self.uidvalidity = 1
+        self.logins = {}
+
+    def _add_login(self, email, password):
+        self.logins[email] = password
+
+    def login(self, email, password):
+        if email not in self.logins or self.logins[email] != password:
+            raise ValidationError
+
+    def logout(self):
+        pass
+
+    def list_folders(self, directory=u'', pattern=u'*'):
+        return []
```

Comment:
I didn't need those things for these tests. If we end up needing those things in the future then we can change it then 😄